nc2zarr/dataslice.py

```diff
@@ -125,7 +125,14 @@ def update_slice(store: Union[str, MutableMapping],
 
     append_dim_var_names = []
     encoding = {}
-    with xr.open_zarr(store) as ds:
+
+    consolidated = True
+    try:
+        _ = zarr.open_consolidated(store)
+    except KeyError:
+        consolidated = False
+
+    with xr.open_zarr(store, consolidated=consolidated) as ds:
```

Comment:
So you open every dataset twice. Why don't you use the result of `zarr.open_consolidated()` in case it succeeds and avoid the second open?

Reply:
The aim is to check whether the dataset is consolidated. Unfortunately neither xarray nor Zarr offer an API function for this, but with Zarr it can be checked using the workaround here: call `open_consolidated`, and catch the `KeyError` which is thrown if the dataset isn't consolidated. If the dataset is consolidated, `open_consolidated` returns a Zarr `Group`, and it's true that we then throw this away and use xarray's `open_zarr`. But there isn't any obvious alternative to this: from this point on, we need an xarray `Dataset` rather than a Zarr `Group`, and xarray doesn't offer any function to create a `Dataset` from a `Group`. `open_zarr` only accepts filesystem paths or `MutableMapping`s.

While there's admittedly a loss of efficiency here, it's not a large one:
1. If the dataset isn't consolidated, the Zarr call fails quickly with only a failed read of the consolidated metadata object.
2. If the dataset is consolidated, it's true that it's opened twice -- but in this case, opening only means parsing the consolidated metadata object. Doing this twice is inefficient, but just reading the consolidated data is relatively fast (by design). The slow operation of opening an *unconsolidated* dataset is never performed twice.

I'll add a comment noting the potential inefficiency and if it proves to be a problem in practice we can implement an optimization, perhaps by reproducing some of the Zarr library's consolidated metadata reading logic in `dataslice` itself. I did consider this implementation too, but thought that since the Zarr format is still evolving it may be best to stick to the APIs for now, at least until Zarr v3 is finalized.